src/lib/commandclass/ColorSwitchCC.ts

```diff
@@ -0,0 +1,762 @@
+			duration: response.duration,
+		};
+	}
+
+	public async set(colorTable: ColorTable): Promise<void> {
+		this.assertSupportsCommand(ColorSwitchCommand, ColorSwitchCommand.Set);
+
+		const cc = new ColorSwitchCCSet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			...colorTable,
+		});
+
+		await this.driver.sendCommand(cc);
+	}
+
+	public async startLevelChange(
+		colorKey: ColorKey,
+		direction: StartLevelChangeDirection,
+		options: StartLevelChangeOptions = {},
+	): Promise<void> {
+		this.assertSupportsCommand(
+			ColorSwitchCommand,
+			ColorSwitchCommand.StartLevelChange,
+		);
+
+		const colorComponent = getColorComponentFromKey(colorKey);
+		if (colorComponent == undefined) {
+			throw new Error(`Unknown color "${colorKey}".`);
+		}
+
+		const cc = new ColorSwitchCCStartLevelChange(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			colorComponent,
+			down: direction === "down",
+			startLevel: options.startLevel,
+		});
+
+		await this.driver.sendCommand(cc);
+	}
+
+	public async stopLevelChange(colorKey: ColorKey): Promise<void> {
+		this.assertSupportsCommand(
+			ColorSwitchCommand,
+			ColorSwitchCommand.StopLevelChange,
+		);
+
+		const colorComponent = getColorComponentFromKey(colorKey);
+		if (colorComponent == undefined) {
+			throw new Error(`Unknown color "${colorKey}".`);
+		}
+
+		const cc = new ColorSwitchCCStopLevelChange(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			colorComponent,
+		});
+
+		await this.driver.sendCommand(cc);
+	}
+
+	protected [SET_VALUE]: SetValueImplementation = async (
+		{ property, propertyKey },
+		value,
+	) => {
+		if (property !== "targetColor") {
+			throwUnsupportedProperty(this.ccId, property);
+		}
+		if (typeof value !== "number") {
+			throwWrongValueType(this.ccId, property, "number", typeof value);
+		}
+
+		if (!propertyKey) {
+			// No error throw util for this?
+			// Might want to treat keyless value as hex "#wwccrrggbb"
+			throw new Error("propertyKey required.");
+		}
+
+		if (typeof propertyKey != "string" || !isColorKey(propertyKey)) {
+			throwUnsupportedPropertyKey(this.ccId, property, propertyKey);
+		}
+
+		await this.set({ [propertyKey]: value });
+
+		// Refresh the current value
+		await this.get(propertyKey);
+	};
+}
+
+@commandClass(CommandClasses["Color Switch"])
+@implementedVersion(3)
+export class ColorSwitchCC extends CommandClass {
+	declare ccCommand: ColorSwitchCommand;
+
+	public async interview(complete: boolean = true): Promise<void> {
+		const node = this.getNode()!;
+		const endpoint = this.getEndpoint()!;
+		const api = endpoint.commandClasses["Color Switch"];
+
+		log.controller.logNode(node.id, {
+			endpoint: this.endpointIndex,
+			message: `${this.constructor.name}: doing a ${
+				complete ? "complete" : "partial"
+			} interview...`,
+			direction: "none",
+		});
+
+		// TODO: Only do this if complete?  Where would we get
```

Comment:
Read them from the value DB (where you stored them). See `BinarySensorCC`, line 142ff.